okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.kt

```diff
@@ -607,5 +637,15 @@ class RealWebSocket(
      * the server doesn't respond the web socket will be canceled.
      */
     private const val CANCEL_AFTER_CLOSE_MILLIS = 60L * 1000
+
+    /**
+     * The smallest message that will be compressed. We use 1024 because smaller messages already
+     * fit comfortably within a single ethernet packet (1500 bytes) even with framing overhead.
+     *
+     * For tests this must be big enough to realize real compression on test messages like
+     * 'aaaaaaaaaa...'. Our tests check if compression was applied just by looking at the size if
+     * the inbound buffer.
+     */
+    const val DEFAULT_MINIMUM_DEFLATE_SIZE = 1024L
```

Comment:
@swankjesse , consider the following case: 

An app is sending a stream of coordinates, JSON encoded. It has dozen of fields (lat, lon, accuracy, heading, timestamp, ...), so the size of the uncompressed JSON is ~110..130B.
With context takeover, a stream of such messages gets compressed quite well (~10-20B per message). 

With all framing overhead, on-wire messages are ~230B uncompressed vs ~120B compressed. These are the ~real numbers we got from Wireshark experiments.

So, even for messages much smaller than 1024B, compression has significant effect on traffic consumption. In my understanding (might be wrong), traffic is measured by the actual IP level traffic, not rounded up to ethernet packet size.

With that, what do you think about making this at least somehow configurable via the OkHttp API? As a user, I'd be surprised to see that compression for my messages (all of which are under 1K) isn't working, despite being implemented in OkHttp.